### PR TITLE
Added tests for dask array applying the Hann window and changed the f…

### PR DESCRIPTION
**Pull request by roxyboy**
I think I'm done with editing this for a while.

### CODE REVIEW

**Comment by rabernat** [comment]
It looks like there is a lot of stuff in this PR. Can you summarize what issues it addresses?

**Comment by roxyboy** [comment]
The main change is that I edited the `_azimuthal_avg` function so that it can take 2D, 3D and 4D data (the averaging is still taken upon two dimensions but the user can now choose the two dimensions, e.g. if the data has x,y,z,t coordinates, it can calculate the isotropic spectrum on any of the two dimensions out of the four). 
However, I couldn't make it use the metadata provided by `xarray` so I pushed the arguments based on `numpy`

File: xrft/xrft.py [hunk 1]
```diff
@@ -243,37 +252,100 @@ def cross_spectrum(da1, da2, dim=None,
...
+                    for i in range(M[1]):
+                        iso_f[j,i] = np.ma.masked_invalid(np.bincount(kidx,
+                                                weights=f.data[j,:,i].ravel())
+                                                        / area) * kr
+            else:
+                for j in range(M[0]):
+                    for i in range(M[1]):
+                        iso_f[j,i] = np.ma.masked_invalid(np.bincount(kidx,
+                                                weights=f.data[j,:,:,i].ravel())
+                                                        / area) * kr
```
Comment (rabernat) [inline xrft/xrft.py]: I can't merge this PR as long as it contains this long `if / else`  section.

There must be a better way to generalize the logic of azimuthal averaging to multiple dimensions. You should only have one call to `np.bincount`. The code above is very difficult to understand and would be hard to maintain.

If you would like to move on with the PR without getting bogged down in this issue, just change `_azimuthal_avg` to raise an error if you attempt to call it on a field with more than two dimensions.
